cosmos/providers/databricks/plugin.py

```diff
@@ -0,0 +1,279 @@
+
+    @provide_session
+    def get_dagrun(self, dag: DAG, run_id, session=None) -> DagRun:
+        """
+        Retrieves the DagRun object associated with the specified DAG and run_id.
+
+        :param dag: The DAG object associated with the DagRun to retrieve.
+        :param run_id: The run_id associated with the DagRun to retrieve.
+        :param session: The SQLAlchemy session to use for the query. If None, uses the default session.
+        :return: The DagRun object associated with the specified DAG and run_id.
+        """
+        return (
+            session.query(DagRun)
+            .filter(DagRun.dag_id == dag.dag_id, DagRun.run_id == run_id)
+            .first()
+        )
+
+    @provide_session
+    def get_ti(self, ti_key: TaskInstanceKey, session=None) -> TaskInstance:
+        return (
+            session.query(TaskInstance)
+            .where(TaskInstance.filter_for_tis([ti_key]))
+            .one()
+        )
+
+
+class DatabricksJobRepairSingleFailedLink(BaseOperatorLink):
+    """Constructs a link to send a repair request for a single databricks task."""
+
+    name = "Repair a single failed task"
+
+    def get_link(
+        self,
+        operator,
+        dttm=None,
+        *,
+        ti_key: TaskInstanceKey | None = None,
+    ) -> str:
+        dag = get_airflow_app().dag_bag.get_dag(ti_key.dag_id)
+        task = dag.get_task(ti_key.task_id)
+        # Should we catch the exception here if there is no return value?
+        metadata = XCom.get_one(
+            task_id=ti_key.task_id,
+            dag_id=ti_key.dag_id,
+            run_id=ti_key.run_id,
+            key="return_value",
+        )
+
+        return (
+            f"/repair_databricks_job?dag_id={ti_key.dag_id}&"
+            f"databricks_conn_id={metadata.databricks_conn_id}&"
+            f"databricks_run_id={metadata.databricks_run_id}&"
+            f"tasks_to_repair={_get_databricks_task_id(task)}"
+        )
+
+
+class RepairDatabricksTasks(AirflowBaseView):
+    default_view = "repair"
+
+    @expose("/repair_databricks_job", methods=["GET"])
+    @has_access(
+        [
+            (permissions.ACTION_CAN_READ, permissions.RESOURCE_WEBSITE),
+        ]
+    )
+    def repair(self):
+        databricks_conn_id, databricks_run_id, dag_id, tasks_to_repair = itemgetter(
+            "databricks_conn_id", "databricks_run_id", "dag_id", "tasks_to_repair"
+        )(request.values)
+        res = _repair_task(
+            databricks_conn_id=databricks_conn_id,
+            databricks_run_id=databricks_run_id,
+            tasks_to_repair=tasks_to_repair,
+        )
+        flash(f"Databricks repair job is starting!: {res}")
+        return redirect(f"/dags/{dag_id}/grid")
+
+
+repair_databricks_view = RepairDatabricksTasks()
+
+repair_databricks_package = {
+    "name": "Test View",
+    "category": "Test Plugin",
+    "view": repair_databricks_view,
+}
+
+
+class CosmosDatabricksPlugin(AirflowPlugin):
+    name = "my_namespace"
```

Comment:
Is this the default and needed value? Or do we need to change it?

Reply:
We can change it I believe.